Sources/ClientRuntime/Networking/Http/Middlewares/DeserializeMiddleware.swift

```diff
@@ -14,12 +25,10 @@ public struct DeserializeMiddleware<Output: HttpResponseBinding,
             Self.MOutput == H.Output,
             Self.Context == H.Context {
 
-            let decoder = context.getDecoder()
             let response = try await next.handle(context: context, input: input) // call handler to get http response
             var copiedResponse = response
             if (200..<300).contains(response.httpResponse.statusCode.rawValue) {
-                let output = try await Output(httpResponse: copiedResponse.httpResponse,
-                                        decoder: decoder)
+                let output = try await httpResponseClosure(response.httpResponse)
```

Comment:
Response closure is called in place of creating an `Output` directly.  (The closure initializes an `Output` value the exact same way though.)

Reply:
This also prevents the middleware from needing to know that `Output` is `HttpResponseBinding` conformant, and allows us to take reference to that protocol out of the entire operation stack.